src/core-services/product/queries/products.js

```diff
@@ -0,0 +1,37 @@
+import applyProductFilters from "../utils/applyProductFilters.js";
+
+/**
+ * @name products
+ * @method
+ * @memberof GraphQL/Products
+ * @summary Query the Products collection for a list of products
+ * @param {Object} context - an object containing the per-request state
+ * @param {Object} input - Request input
+ * @param {Boolean} [isArchived] - Filter by archived
+ * @param {Boolean} [isVisible] - Filter by visibility
+ * @param {String} [metafieldKey] - Filter by metafield key
+ * @param {String} [metafieldValue] - Filter by metafield value
+ * @param {Number} [priceMax] - Filter by price range maximum value
+ * @param {Number} [priceMin] - Filter by price range minimum value
+ * @param {String[]} [productIds] - List of product IDs to filter by
+ * @param {String} [query] - Regex match query string
+ * @param {String[]} shopIds - List of shop IDs to filter by
+ * @param {String[]} [tagIds] - List of tag ids to filter by
+ * @returns {Promise<Object>} Products object Promise
+ */
+export default async function products(context, input) {
+  const { checkPermissions, collections } = context;
+  const { Products } = collections;
+  const productFilters = input;
+
+  // Check the permissions for all shop requested
+  await Promise.all(productFilters.shopIds.map(async (shopId) => (
+    checkPermissions(["owner", "admin", "createProduct"], shopId)
```

Comment:
We already have `product/admin` if you want to use that, otherwise creating `product/read` works.

That being said, these will all be different very soon anyway so it doesn't _really_ matter, as it'll all be refactored in https://github.com/reactioncommerce/reaction/pull/5772, which we hope to have merged into `release-3.0.0` soon.

Reply:
Then let's leave it alone as the check is going to change drastically anyway.

Reply:
@kieckhafer @rosshadden @aldeed @ticean 
This is an important conversation to have, re: standards and policies for permission checks, I agree with @mikemurray that if we're about to refactor this all anyway, this ticket may not be the best place or time to have this conversation.

Do we have any documentation started for how our team should think about adding permision checks after #5772? What about for other services?

Reply:
Working on documentation in the (upcoming) authorization plugin. We do outline the standards there, we're just in a weird transition of the `release-3.0.0` branch being worked on while we are refactoring all of this. It shouldn't be an issue in a few days / weeks once our auth branch is merged, at that time we'll have the new standards all laid out and all over the code.